### DISCUSSION TITLE
[求助] iStoreOS 安装 AdGuardHome 后无法拦截 Android 上需被过滤的域名

### DISCUSSION THREAD

**Discussion by fejeIaqbhbv** [istoreos/istoreos#1871] [General]
不知道有没有遇到过的大佬提供下解决办法

我的(~最简~)复现步骤：
1. 刷入 [iStoreOS](https://sourceforge.net/projects/istoreos/files/iStoreOS/x86_64_efi/istoreos-22.03.7-2024102509-x86-64-squashfs-combined-efi.img.gz/download)
2. 安装 [AdGuardHome](https://github.com/AUK9527/Are-u-ok/raw/main/x86/all/adguardhome.run)
3. 运行 AdGuardHome （3种重定向模式都能复现，即dnsmasq作为上游、重定向53到AdGuardHome端口、用AdGuardHome替代53端口）
4. 添加拦截规则 `||baidu.com^`（如果原本的上游dns不好用也可以顺便换了 `https://dns.alidns.com/dns-query` `https://doh.pub/dns-query`）不进行其它设置项变更
5. 重启 iStoreOS 、电脑、Android 设备以释放缓存
6. 电脑访问 `www.baidu.com` 发现无法访问 查看 AdGuardHome 控制面板确认有拦截
7. Android 设备访问 `www.baidu.com` 提示无法访问，但强制刷新网页就可以正常加载，电脑无法复现。查看 AdGuardHome 控制面板确认也有对应的拦截记录，但还是可以正常访问。

只是用百度举例子，并不只有百度会出问题
如果需要真正的广告网站可以用`||umeng.com^`，该网址在不少在线规则中都存在。

已确认dns和网关指向均无误，
已使用`3,192.168.100.1` `6,192.168.100.1`dhcp代码强制指定设备
已尝试手机切飞行模式或拔 sim 卡只用 WiFi
准备了一部手机还原出厂设置后访问被拦截网站，但也是刷新后就可以正常加载

谢谢

**Comment by jjm2473** [comment]
可能用了安全DNS吧，换成不同浏览器试试

**Comment by fejeIaqbhbv** [comment]
感谢回复

和浏览器无关

Android系统内置了备用dns 8.8.4.4 ，学习了怎么抓包然后通过 `getprop | grep dns` 查到的

然后这个路由器上游是一台开了代理的，就这么巧......

目前把AdGuardHome LuCI提供的防火墙规则 `iptables -t nat -A PREROUTING -p udp -d 192.168.100.1 --dport 53 -j REDIRECT --to-port 5353` 换成了 `iptables -t nat -I PREROUTING -p udp --dport 53 -j REDIRECT --to-port 5353` 先用着

---

再补一个`iptables -t nat -I OUTPUT -p udp --dport 53 -j REDIRECT --to-port 5353`，否则路由器本机上跑的应用没法从adghome拿dns，配置上游dns时记得用非53端口的dohdotdoq...

**Reply by jjm2473** [reply]
新的固件可以在“网络”-“DHCP/DNS”-“常规”设置里，勾选“DNS 重定向”。
